pilot/pkg/xds/debug.go

```diff
@@ -232,7 +232,11 @@ func (s *DiscoveryServer) registryz(w http.ResponseWriter, req *http.Request) {
 	}
 	_, _ = fmt.Fprintln(w, "[")
 	for _, svc := range all {
-		b, err := json.MarshalIndent(svc, "", "  ")
+		b, err := func() ([]byte, error) {
+			svc.Mutex.RLock()
```

Comment:
In Envoy/c++ we tend to try and stick to RAII, so this would probably translate into a ScopedMutex or some such. Is there a similar construct we can use here?